File: backend/app/schemas/planning.py
from pydantic import BaseModel, Field


class ImageResource(BaseModel):
    category: str
    url: str
    thumbnail_url: str
    alt: str
    provider: str
    placeholder: bool = True
    source_url: str | None = None
    license: str | None = None
    credit: str | None = None


class DestinationRecommendationRequest(BaseModel):
    departure_city: str | None = None
    budget_min: int | None = Field(default=None, ge=0)
    budget_max: int | None = Field(default=None, ge=0)
    duration_days: int = Field(default=4, ge=1, le=30)
    season: str | None = None
    travel_style: list[str] = Field(default_factory=list)
    interests: list[str] = Field(default_factory=list)


class DestinationItem(BaseModel):
    id: str
    name: str
    country_or_region: str
    match_score: int
    budget_range: str
    best_season: str
    vibe_tags: list[str]
    reasons: list[str]
    hero_image: ImageResource
    gallery: list[ImageResource]


class DestinationRecommendationPayload(BaseModel):
    query_summary: str
    destinations: list[DestinationItem]


class RouteGenerationRequest(BaseModel):
    destination_id: str | None = None
    destination_name: str
    duration_days: int = Field(default=4, ge=1, le=30)
    pace: str = "balanced"
    travelers: int = Field(default=1, ge=1, le=20)
    interests: list[str] = Field(default_factory=list)


class RouteSpot(BaseModel):
    time_slot: str
    name: str
    description: str
    suggested_duration_hours: float
    category: str | None = None
    address: str | None = None
    latitude: float | None = None
    longitude: float | None = None
    images: list[ImageResource]


class RouteDayPlan(BaseModel):
    day: int
    theme: str
    commute_tip: str
    spots: list[RouteSpot]


class RouteOption(BaseModel):
    id: str
    title: str
    pace: str
    estimated_budget: str
    photo_score: float
    summary: str
    days: list[RouteDayPlan]


class RouteGenerationPayload(BaseModel):
    destination_name: str
    options: list[RouteOption]


class MediaPlaceholderRequest(BaseModel):
    categories: list[str] = Field(default_factory=list)
    destination_name: str | None = None
    keywords: list[str] = Field(default_factory=list)


class MediaPlaceholderGroup(BaseModel):
    category: str
    items: list[ImageResource]


class MediaPlaceholderPayload(BaseModel):
    destination_name: str | None = None
    assets: list[MediaPlaceholderGroup]


class DestinationWeatherPayload(BaseModel):
    destination_name: str
    available: bool
    source: str = "amap"
    city: str | None = None
    adcode: str | None = None
    weather: str | None = None
    temperature: str | None = None
    wind_direction: str | None = None
    wind_power: str | None = None
    humidity: str | None = None
    report_time: str | None = None
    message: str | None = None

File: frontend/src/types/planning.ts
/**
 * 规划相关类型定义
 * 与后端 `app/schemas/planning.py` 对齐
 */

/** 图片资源 */
export interface ImageResource {
  category: string
  url: string
  thumbnail_url: string
  alt: string
  provider: string
  placeholder: boolean
  source_url?: string | null
  license?: string | null
  credit?: string | null
}

/** 目的地推荐请求 */
export interface DestinationRecommendationRequest {
  departure_city?: string | null
  budget_min?: number | null
  budget_max?: number | null
  duration_days?: number
  season?: string | null
  travel_style?: string[]
  interests?: string[]
}

/** 目的地推荐项 */
export interface DestinationItem {
  id: string
  name: string
  country_or_region: string
  match_score: number
  budget_range: string
  best_season: string
  vibe_tags: string[]
  reasons: string[]
  hero_image: ImageResource
  gallery: ImageResource[]
}

/** 目的地推荐响应数据 */
export interface DestinationRecommendationPayload {
  query_summary: string
  destinations: DestinationItem[]
}

/** 路线生成请求 */
export interface RouteGenerationRequest {
  destination_id?: string | null
  destination_name: string
  duration_days?: number
  pace?: string
  travelers?: number
  interests?: string[]
}

/** 路线中的景点 */
export interface RouteSpot {
  time_slot: string
  name: string
  description: string
  suggested_duration_hours: number
  category?: string | null
  address?: string | null
  latitude?: number | null
  longitude?: number | null
  images: ImageResource[]
}

/** 路线中的一天计划 */
export interface RouteDayPlan {
  day: number
  theme: string
  commute_tip: string
  spots: RouteSpot[]
}

/** 路线方案 */
export interface RouteOption {
  id: string
  title: string
  pace: string
  estimated_budget: string
  photo_score: number
  summary: string
  days: RouteDayPlan[]
}

/** 路线生成响应数据 */
export interface RouteGenerationPayload {
  destination_name: string
  options: RouteOption[]
}

/** 媒体占位请求 */
export interface MediaPlaceholderRequest {
  categories?: string[]
  destination_name?: string | null
  keywords?: string[]
}

/** 媒体占位分组 */
export interface MediaPlaceholderGroup {
  category: string
  items: ImageResource[]
}

/** 媒体占位响应数据 */
export interface MediaPlaceholderPayload {
  destination_name: string | null
  assets: MediaPlaceholderGroup[]
}
